packages/devtools_app/lib/src/screens/deep_link_validation/deep_links_screen.dart

```diff
@@ -24,8 +130,112 @@ class DeepLinksScreen extends Screen {
 
   @override
   Widget build(BuildContext context) {
-    return const Center(
-      child: Text('TODO: build deep link validation tool'),
+    return const DeepLinkPage();
+  }
+}
+
+class DeepLinkPage extends StatefulWidget {
+  const DeepLinkPage({super.key});
+
+  @override
+  State<DeepLinkPage> createState() => _DeepLinkPageState();
+}
+
+class _DeepLinkPageState extends State<DeepLinkPage> {
+  String responseBody = 'empty response';
+  int? selectedRowIndex;
+  String searchContent = '';
+  bool bundleByDomain = false;
+
+  void setBundleByDomain(bool shouldBundleByDomain) {
+    setState(() {
+      bundleByDomain = shouldBundleByDomain;
+    });
+  }
+
+  void setResponse(String response) {
+    setState(() {
+      responseBody = response;
+    });
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return ListView(
+      children: [
+        Row(
+          children: [
+            Expanded(child: buildTitle('All deep links', context)),
+            InkWell(
+              onTap: () {
+                setBundleByDomain(!bundleByDomain);
+              },
+              child: const Text('Bundle by domain'),
+            ),
+            const SizedBox(width: 10),
+            SearchBar(
+              leading: const Icon(Icons.search),
+              onChanged: (value) {
+                setState(() {
+                  searchContent = value.toLowerCase();
+                });
+              },
+              constraints: BoxConstraints.tight(const Size(200, 40)),
```

Comment:
please use `defaultSearchFieldWidth` and `defaultTextFieldHeight` from `theme.dart`. Many consts in theme.dart are also wrapped with `scaleByFontFactor` which ensures your UI will scale nicely as the font factor changes